static/js/confirm_dialog.js

```diff
@@ -74,7 +90,9 @@ export function launch(conf) {
 
     // Set up handlers.
     yes_button.on("click", () => {
-        overlays.close_modal("#confirm_dialog_modal");
+        if (!conf.loading_spinner) {
+            overlays.close_modal("#confirm_dialog_modal");
+        }
         conf.on_click();
```

Comment:
Would it be a better API to call `show_confirm_dialog_spinner` and then pass `hide_confirm_dialog_spinner` to the `on_click` method as a callback that it can use when appropriate?

Reply:
SGTM, but don't think we should pass `hide_confirm_dialog_spinner` to `on_click`. Calling it directly looks better. (more "functional")